`test_parse_config` failing in openSUSE Tumbleweed

### mkosi commit the issue has been seen with

v24

### Used host distribution

openSUSE Tumbleweed

### Used target distribution

_No response_

### Linux kernel version used

_No response_

### CPU architectures issue was seen on

x86_64

### Unexpected behaviour you saw

Seen in OBS, full log here: https://build.opensuse.org/package/live_build_log/home:afeijoo:branches:Virtualization/mkosi/openSUSE_Tumbleweed/x86_64

### Used mkosi config

_No response_

### mkosi output

```sh
[   25s] =================================== FAILURES ===================================
[   25s] ______________________________ test_parse_config _______________________________
[   25s] 
[   25s] tmp_path = PosixPath('/tmp/pytest-of-abuild/pytest-0/test_parse_config0')
[   25s] 
[   25s]     def test_parse_config(tmp_path: Path) -> None:
[   25s]         d = tmp_path
[   25s]     
[   25s]         (d / "mkosi.conf").write_text(
[   25s]             """\
[   25s]             [Distribution]
[   25s]             Distribution=ubuntu
[   25s]             Architecture=arm64
[   25s]             Repositories=epel,epel-next
[   25s]     
[   25s]             [Content]
[   25s]             Packages=abc
[   25s]             Environment=MY_KEY=MY_VALUE
[   25s]     
[   25s]             [Output]
[   25s]             Format=cpio
[   25s]             ImageId=base
[   25s]     
[   25s]             [Host]
[   25s]             Credentials=my.cred=my.value
[   25s]             """
[   25s]         )
[   25s]     
[   25s]         with chdir(d):
[   25s]             _, [config] = parse_config()
[   25s]     
[   25s]         assert config.distribution == Distribution.ubuntu
[   25s]         assert config.architecture == Architecture.arm64
[   25s]         assert config.packages == ["abc"]
[   25s]         assert config.output_format == OutputFormat.cpio
[   25s]         assert config.image_id == "base"
[   25s]     
[   25s]         with chdir(d):
[   25s]             _, [config] = parse_config(
[   25s]                 [
[   25s]                     "--distribution", "fedora",
[   25s]                     "--environment", "MY_KEY=CLI_VALUE",
[   25s]                     "--credential", "my.cred=cli.value",
[   25s]                     "--repositories", "universe",
[   25s]                 ]
[   25s]             )
[   25s]     
[   25s]         # Values from the CLI should take priority.
[   25s]         assert config.distribution == Distribution.fedora
[   25s]         assert config.environment["MY_KEY"] == "CLI_VALUE"
[   25s]         assert config.credentials["my.cred"] == "cli.value"
[   25s]         assert config.repositories == ["epel", "epel-next", "universe"]
[   25s]     
[   25s]         with chdir(d):
[   25s]             _, [config] = parse_config(
[   25s]                 [
[   25s]                     "--distribution", "",
[   25s]                     "--environment", "",
[   25s]                     "--credential", "",
[   25s]                     "--repositories", "",
[   25s]                 ]
[   25s]             )
[   25s]     
[   25s]         # Empty values on the CLIs resets non-collection based settings to their defaults and collection based settings to
[   25s]         # empty collections.
[   25s] >       assert config.distribution == detect_distribution()[0]
[   25s] E       AssertionError: assert <Distribution.custom: 'custom'> == None
[   25s] E        +  where <Distribution.custom: 'custom'> = Config(profile=None, include=[], initrd_include=[], dependencies=[], minimum_version=None, pass_environment=[], distribution=<Distribution.custom: 'custom'>, release='', architecture=<Architecture.arm64: 'arm64'>, mirror=None, local_mirror=None, repository_key_check=True, repositories=[], cacheonly=<Cacheonly.auto: 'auto'>, package_manager_trees=[], output_format=<OutputFormat.cpio: 'cpio'>, manifest_format=[], output='base', compress_output=<Compression.zstd: 'zstd'>, compress_level=3, output_dir=None, workspace_dir=None, cache_dir=None, package_cache_dir=None, build_dir=None, image_id='base', image_version=None, split_artifacts=False, repart_dirs=[], sector_size=None, repart_offline=True, overlay=False, use_subvolumes=<ConfigFeature.auto: 'auto'>, seed=UUID('876cdbea-8953-4abb-a18c-af7d7b989f61'), packages=['abc'], build_packages=[], volatile_packages=[], package_directories=[], volatile_package_directories=[], with_recommends=False, with_docs=True, base_trees=[], skeleton_trees=[], extra_trees=[], remove_packages=[], remove_files=[], clean_package_metadata=<ConfigFeature.auto: 'auto'>, source_date_epoch=1722246072, configure_scripts=[], sync_scripts=[], prepare_scripts=[], bui...erm.hvc0=dumb', 'systemd.tty.columns.hvc0=80', 'systemd.tty.rows.hvc0=24', 'ip=enc0:any', 'ip=enp0s1:any', 'ip=enp0s2:any', 'ip=host0:any', 'ip=none', 'loglevel=4', 'SYSTEMD_SULOGIN_FORCE=1', 'systemd.tty.term.console=dumb', 'systemd.tty.columns.console=80', 'systemd.tty.rows.console=24', 'console=hvc0', 'TERM=dumb'], acl=False, tools_tree=None, tools_tree_distribution=<Distribution.custom: 'custom'>, tools_tree_release='', tools_tree_mirror=None, tools_tree_repositories=[], tools_tree_package_manager_trees=[], tools_tree_packages=[], tools_tree_certificates=True, runtime_trees=[], runtime_size=None, runtime_scratch=<ConfigFeature.auto: 'auto'>, runtime_network=<Network.user: 'user'>, runtime_build_sources=False, unit_properties=[], ssh_key=None, ssh_certificate=None, machine=None, forward_journal=None, vmm=<Vmm.qemu: 'qemu'>, qemu_gui=False, qemu_smp=1, qemu_mem=2147483648, qemu_kvm=<ConfigFeature.auto: 'auto'>, qemu_vsock=<ConfigFeature.auto: 'auto'>, qemu_vsock_cid=<QemuVsockCID.auto: -1>, qemu_swtpm=<ConfigFeature.auto: 'auto'>, qemu_cdrom=False, qemu_firmware=<QemuFirmware.auto: 'auto'>, qemu_firmware_variables=None, qemu_kernel=None, qemu_drives=[], qemu_args=[], image=None).distribution
[   25s] 
[   25s] tests/test_config.py:154: AssertionError
[   25s] =========================== short test summary info ============================
[   25s] FAILED tests/test_config.py::test_parse_config - AssertionError: assert <Dist...
[   25s] ================= 1 failed, 245 passed, 19 deselected in 8.97s =================
```

--- a/tests/test_config.py
+++ b/tests/test_config.py
@@ -151,7 +151,7 @@ def test_parse_config(tmp_path: Path) -> None:
 
     # Empty values on the CLIs resets non-collection based settings to their defaults and collection based settings to
     # empty collections.
-    assert config.distribution == detect_distribution()[0]
+    assert config.distribution == (detect_distribution()[0] or Distribution.custom)
     assert "MY_KEY" not in config.environment
     assert "my.cred" not in config.credentials
     assert config.repositories == []

